Agents/DistrictHeatingAgent/DistrictHeatingAgent/src/main/java/uk/ac/cam/cares/jps/agent/heat/HeatNetworkInputAgent.java

```diff
@@ -0,0 +1,679 @@
+    public static String Aussentemperatur="";
+    public static String Waermeeinspeisung="";
+    public static String MHKWTempRuecklauf="";
+
+    // The instantiation of static data
+    public void dataInstantiation() {
+
+        String staticData = System.getenv("CSVPATH");
+        String[] numericValue = ReadColInput(1,staticData,",");
+
+        // static data
+        float Value_VolumetricFlowRate = Float.parseFloat(numericValue[0]);
+        float Value_RatedElectricalPower = Float.parseFloat(numericValue[1]);
+        float Value_MinimumThermalLoad = Float.parseFloat(numericValue[2]);
+        float Value_MinimumIdleTime = Float.parseFloat(numericValue[3]);
+        float Value_MinPurchase = Float.parseFloat(numericValue[4]);
+        float Value_MaxPurchase = Float.parseFloat(numericValue[5]);
+        float Value_TierOneCap = Float.parseFloat(numericValue[6]);
+        float Value_TierTwoCap = Float.parseFloat(numericValue[7]);
+        float Value_TierOneUnitPrice = Float.parseFloat(numericValue[8]);
+        float Value_TierTwoUnitPrice = Float.parseFloat(numericValue[9]);
+        float Value_ThermalLoad_GT = Float.parseFloat(numericValue[10]);
+        float Value_HCV_GT = Float.parseFloat(numericValue[11]);
+        float Value_LCV_GT = Float.parseFloat(numericValue[12]);
+        float Value_CO2Factor_GT = Float.parseFloat(numericValue[13]);
+        float Value_ThermalLoad_HB4 = Float.parseFloat(numericValue[14]);
+        float Value_HCV_HB4 = Float.parseFloat(numericValue[15]);
+        float Value_LCV_HB4 = Float.parseFloat(numericValue[16]);
+        float Value_CO2Factor_HB4 = Float.parseFloat(numericValue[17]);
+        float Value_ThermalLoad_HB5 = Float.parseFloat(numericValue[18]);
+        float Value_HCV_HB5 = Float.parseFloat(numericValue[19]);
+        float Value_LCV_HB5 = Float.parseFloat(numericValue[20]);
+        float Value_CO2Factor_HB5 = Float.parseFloat(numericValue[21]);
+        float Value_ThermalLoad_HB6 = Float.parseFloat(numericValue[22]);
+        float Value_HCV_HB6 = Float.parseFloat(numericValue[23]);
+        float Value_LCV_HB6 = Float.parseFloat(numericValue[24]);
+        float Value_CO2Factor_HB6 = Float.parseFloat(numericValue[25]);
+
+
+        // For the Consumer part
+        UpdateBuilder Consumer_ub =
+                new UpdateBuilder()
+                        .addInsert(NodeFactory.createURI(KB + "Consumer"), NodeFactory.createURI(OHN + "hasHeatDemand"), NodeFactory.createURI(KB + "HeatDemand"))
+                        .addInsert(NodeFactory.createURI(KB + "Consumer"), NodeFactory.createURI(RDF_TYPE), NodeFactory.createURI(OHN + "Consumer"))
+                        .addInsert(NodeFactory.createURI(KB + "HeatDemand"), NodeFactory.createURI(RDF_TYPE), NodeFactory.createURI(OHN + "HeatDemand"));
+        UpdateRequest Consumer_ur = Consumer_ub.buildRequest();
+        AccessAgentCaller.updateStore("http://host.docker.internal:48888/ontoheatnet", Consumer_ur.toString());
```

Comment:
Instead of putting the same URL string literal in 14 different places, please use a variable instead. That way, if you ever need to change it, you only need to change it in one place, rather than 14.

Secondly, you've already got a client.properties file in which one can configure query and update URLs for a SPARQL endpoint. Why don't you make use of that, instead of hard-coding the URL? Otherwise, what's the point of the setting in the client.properties file?

Thirdly, if you impose a URL, there is no point in using the access agent at all. You may want to ask, e.g., Hou Yee to explain how he does it in his instantiation agents.